src/alarms/notifications/NewNotification.js

```diff
@@ -245,33 +245,44 @@ class NewNotification extends Component {
       comparison,
       messages,
       raster,
-      markerPosition
+      markerPosition,
+      timeseriesUuid
     } = this.state;
 
-    fetch("/api/v3/rasteralarms/", {
+    let url = "";
```

Comment:
Ik vind deze functie veel te lang.

Een functie die een klik afhandelt moet niet weten dat er verschillende soorten alarms zijn, hoe je het verschil kunt zien in de state, hoe hun JSON er uit ziet, wat de URLs zijn. Dat soort dingen moeten op 1 plek staan en een click handler is daar niet de logische plek voor.

Wel dat er na het saven een notificatie moet komen en de pagina verandert.

Ik zie iets voor me als een functie `this.selectedAlarmType()`, en functies ergens buiten de React component die saveTimeseriesAlarm en saveRasterAlarm doen, en een promise met JSON opleveren?

Of vind jij dat het dan juist ingewikkelder wordt?

Reply:
@remcogerlich 
Hi Remco,
Heb de functie naar een andere file gezet.
Is dit meer wat je in gedachte had?
groet Tom

Reply:
Nee, eigenlijk niet, hij doet nog steeds evenveel. Ik zal morgenochtend in de trein eens kijken of ik het meer in de richting die ik voor me zie kan duwen, dan kunnen we eens vergelijken.